assets/src/settings-page/template-modes.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import {
+	THEME_IS_SUPPORTED,
+	THEME_SUPPORT_ARGS,
+	THEME_SUPPORTS_READER_MODE,
+} from 'amp-settings';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { TemplateModeOption } from '../components/template-mode-option';
+import '../css/template-mode-selection.css';
+import { AMPNotice, NOTICE_SIZE_LARGE, NOTICE_TYPE_INFO, NOTICE_TYPE_WARNING } from '../components/amp-notice';
+import { Options } from '../components/options-context-provider';
+
+/**
+ * Provides the notice to show in the reader theme support mode selection.
+ *
+ * @param {string} themeSupport The current theme support mode.
+ */
+function getReaderNotice( themeSupport ) {
+	switch ( true ) {
+		// Theme has built-in support or has declared theme support with the paired flag false.
+		case 'reader' === themeSupport && ( THEME_IS_SUPPORTED || ( 'object' === typeof THEME_SUPPORT_ARGS && false === THEME_SUPPORT_ARGS.paired ) ):
+			return (
+				<AMPNotice size={ NOTICE_SIZE_LARGE } type={ NOTICE_TYPE_WARNING }>
+					<p>
+						{ __( 'Your active theme is known to work well in standard mode.', 'amp' ) }
+					</p>
+				</AMPNotice>
+			);
+
+		// Theme has built-in support or has declared theme support with the paired flag true.
+		case 'reader' === themeSupport && ( THEME_IS_SUPPORTED || ( 'object' === typeof THEME_SUPPORT_ARGS && true === THEME_SUPPORT_ARGS.paired ) ):
```

Comment:
At this point we don't care what what `THEME_SUPPORT_ARGS.paired` is. All we care is that the theme declares `amp` theme support _or_ a core theme is active:

```suggestion
		case 'reader' === themeSupport && ( THEME_IS_SUPPORTED || false !== THEME_SUPPORT_ARGS ):
```